### PR TITLE
Dialog util

### PR DESCRIPTION
**Pull request by btkelly**
- Added dialog util that can be used to easily show dialogs related to the bootstrap check.
- Added a simple logging class that still needs the LogLevel connected and set from the Gandalf installer.
- Added the ability to set the package name from the Gandalf installer

### CODE REVIEW

**Comment by stkent** [comment]
Does this PR aim to completely resolve #6?

File: example/src/main/java/io/github/btkelly/gandalf/example/activities/SplashActivity.java [hunk 1]
```diff
@@ -49,6 +49,13 @@ public int contentView() {
     //** Methods below are just for allowing switching of bootstrap file, not related to Gandalf implementation **//
 
     @Override
+    protected void onStop() {
+        super.onStop();
+        MockWebServerUtil.setMockBootstrapRes(this, R.raw.no_action_bootstrap);
+        System.exit(0);
```
Comment (stkent) [inline example/src/main/java/io/github/btkelly/gandalf/example/activities/SplashActivity.java]: What's going on here? What's the effect of this call?

**Comment by btkelly** [comment]
I would say yes, but as you can see in the code I changed it a bit from my original requirement of it being dumb. Seemed like a better interaction to have it save the user responses and react to clicks.

File: example/src/main/java/io/github/btkelly/gandalf/example/utils/MockWebServerUtil.java [hunk 2]
```diff
@@ -65,7 +67,7 @@ public void run() {
                     MockResponse mockResponse = new MockResponse();
                     mockResponse.setResponseCode(200);
                     mockResponse.setBody(mockBootstrapJsonBody);
-                    mockResponse.setBodyDelay(4, TimeUnit.SECONDS);
+                    mockResponse.setBodyDelay(2, TimeUnit.SECONDS);
```
Comment (stkent) [inline example/src/main/java/io/github/btkelly/gandalf/example/utils/MockWebServerUtil.java]: What factors influence a good choice for this number?

File: example/src/main/java/io/github/btkelly/gandalf/example/activities/SplashActivity.java (hunk 1 above)
Comment (btkelly) [inline example/src/main/java/io/github/btkelly/gandalf/example/activities/SplashActivity.java]: This is to fix the issue where you are stuck with the force update call as you are always blocked when trying to get into the app. This allows you to see each scenario once and on restart it will revert back to no action where you can again select which one to simulate.

File: example/src/main/java/io/github/btkelly/gandalf/example/App.java [hunk 3]
```diff
@@ -31,8 +32,11 @@ public void onCreate() {
 
         String mockBootstrapUrl = MockWebServerUtil.startMockWebServer(this);
 
+        Log.d("GandalfExample", "Mock server started at " + mockBootstrapUrl);
```
Comment (stkent) [inline example/src/main/java/io/github/btkelly/gandalf/example/App.java]: Should this raw log statement be here?

File: example/src/main/java/io/github/btkelly/gandalf/example/utils/MockWebServerUtil.java (hunk 2 above)
Comment (btkelly) [inline example/src/main/java/io/github/btkelly/gandalf/example/utils/MockWebServerUtil.java]: Nothing really, I just wanted to show that what could be done like a progress spinner while the call is loading. 4 seemed too long while I was testing, got sick of waiting.

File: example/src/main/java/io/github/btkelly/gandalf/example/App.java (hunk 3 above)
Comment (btkelly) [inline example/src/main/java/io/github/btkelly/gandalf/example/App.java]: I did leave that there as it's only part of the example app and didn't think that should have a log level type setting or a dependency on a special logging class for different build types.

File: gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java [hunk 4]
```diff
@@ -106,6 +131,7 @@ public void onSuccess(Bootstrap bootstrap) {
 
                     @Override
                     public void onError(IOException e) {
+                        LoggerUtil.logE("Error fetching bootstrap: " + e.getMessage());
                         //In any error case we should let the user in as to not block based on a bug
```
Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java]: We may want to make this configurable by the library consumer?

Comment (btkelly) [inline gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java]: Yup, that's why I added #29, figured it could be it's own PR

Comment (btkelly) [inline gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java]: And didn't know if we wanted something like Amplify has where the logger is injected all over instead of static calls

File: gandalf/src/main/java/io/github/btkelly/gandalf/activities/GandalfActivity.java [hunk 5]
```diff
@@ -69,32 +67,40 @@ protected final void onResume() {
...
     }
 
     @Override
     public final void onNoActionRequired() {
         youShallPass();
     }
+
+    @Override
+    public void continueLoading() {
+        youShallPass();
```
Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/activities/GandalfActivity.java]: Do we need separate callbacks for `continueLoading` and `onNoActionRequired`? Should the library consumer have to care about whether there was no prompt or the prompt was declined?

File: gandalf/src/main/java/io/github/btkelly/gandalf/utils/PlayStoreUtil.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+public final class PlayStoreUtil {
+
+    private static final String ANDROID_MARKET_URI_PREFIX = "market://details?id=";
+    private static final String GOOGLE_PLAY_STORE_URI_PREFIX = "https://play.google.com/store/apps/details?id=";
+
+    private PlayStoreUtil() {
+
+    }
+
+    public static void openPlayStoreToRate(@Nullable final Activity activity, @Nullable final String packageName) {
```
Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/utils/PlayStoreUtil.java]: Method needs a rename.

File: gandalf/src/main/java/io/github/btkelly/gandalf/activities/GandalfActivity.java (hunk 5 above)
Comment (btkelly) [inline gandalf/src/main/java/io/github/btkelly/gandalf/activities/GandalfActivity.java]: It's that they are different events, one is the no updates and another is the user declined an update or alert. This is so a consumer can use the GandalfCallback with or without a dialog.

File: gandalf/src/main/java/io/github/btkelly/gandalf/utils/PlayStoreUtil.java (hunk 6 above)
Comment (btkelly) [inline gandalf/src/main/java/io/github/btkelly/gandalf/utils/PlayStoreUtil.java]: Suggestions?

File: gandalf/src/main/java/io/github/btkelly/gandalf/utils/PlayStoreUtil.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import static android.content.Intent.ACTION_VIEW;
+
+public final class PlayStoreUtil {
+
+    private static final String ANDROID_MARKET_URI_PREFIX = "market://details?id=";
+    private static final String GOOGLE_PLAY_STORE_URI_PREFIX = "https://play.google.com/store/apps/details?id=";
```
Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/utils/PlayStoreUtil.java]: Based on the minsdk of your lib, you probably don't need to check both of these any more. Pretty sure I changed it in amplify, if you want to take a peek.

File: example/src/main/java/io/github/btkelly/gandalf/example/activities/SplashActivity.java (hunk 1 above)
Comment (stkent) [inline example/src/main/java/io/github/btkelly/gandalf/example/activities/SplashActivity.java]: Ah, smart! Might be worth throwing a comment in here with that^ explanation.

File: example/src/main/java/io/github/btkelly/gandalf/example/utils/MockWebServerUtil.java (hunk 2 above)
Comment (stkent) [inline example/src/main/java/io/github/btkelly/gandalf/example/utils/MockWebServerUtil.java]: Ah, makes sense!

File: example/src/main/java/io/github/btkelly/gandalf/example/App.java (hunk 3 above)
Comment (stkent) [inline example/src/main/java/io/github/btkelly/gandalf/example/App.java]: Ah, I see!

File: gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java (hunk 4 above)
Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java]: I'll probably revert that at some point. Neither the amplify nor the bugshaker loggers are ideal yet imo.

File: gandalf/src/main/java/io/github/btkelly/gandalf/activities/GandalfActivity.java (hunk 5 above)
Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/activities/GandalfActivity.java]: Not sure I follow, what is the use-case here?

File: gandalf/src/main/java/io/github/btkelly/gandalf/utils/PlayStoreUtil.java (hunk 6 above)
Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/utils/PlayStoreUtil.java]: `openPlayStoreToUpdate` seems logical to me!

File: gandalf/src/main/java/io/github/btkelly/gandalf/activities/GandalfActivity.java (hunk 5 above)
Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/activities/GandalfActivity.java]: Discussed in Slack; decided we can assess the value of separate cases more easily when the library has matured a bit.

**Comment by stkent** [comment]
Approved
